Firestore/Example/Tests/Util/FSTHelpers.h

```diff
@@ -148,6 +149,31 @@ inline NSString *FSTRemoveExceptionPrefix(NSString *exception) {
     XCTAssertTrue(didThrow, ##__VA_ARGS__);                             \
   } while (0)
 
+/** An implementation of FSTTargetMetadataProvider that can be customized at construction time. */
+@interface FSTTestTargetMetadataProvider : NSObject <FSTTargetMetadataProvider>
+
+/**
+ * Creates an FSTTestTargetMetadataProvider that returns the given document for the existing key
+ * callback (via `remoteKeysForTarget`) and additionally marks the given targets as active (via
+ * `queryDataForTarget`).
```

Comment:
This comment is helpful but it assumes you understand everything about how the system works. It would be even more helpful if you framed this in higher-level terms of what this represents in a user visible way or in terms of what history of events we're simulating. I have no idea if it's accurate or not, but something like this:

> Creates an `FSTTestTargetMetadataProvider` that behaves as if there's an established listen for each of the given `targets`, where each target has already seen query results containing just the given `documentKey`.
> 
> Internally this means that the `remoteKeysForTarget` callback for these targets will return the `documentKey` and that targets will be returned as active from the `queryDataForTarget` target.

Similar feedback applies below.

Reply:
Done (I mostly went with your comment).